### DISCUSSION TITLE
Change in Guidance on Committing Lockfiles

### DISCUSSION THREAD

**Discussion by rusty-snake** [libseccomp-rs/libseccomp-rs#241] [General]
https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html
https://doc.rust-lang.org/nightly/cargo/faq.html#why-have-cargolock-in-version-control

**Comment by ManaSugi** [comment]
@rusty-snake Thanks for the information.

In my understanding, even if we commit `Cargo.lock` in our repository and publish our crate,  when users install and build our library from `crates.io`, which means that they specifiy our crate in their `Cargo.toml`,  the `Cargo.lock` is always ignored.
(In case of binary crate, `cargo install` ignores `Cargo.lock` by default but it can be enabled with `cargo install --locked`.)

Hence, this discussion would be whether we should commit `Cargo.lock` in order to do debug easily on our side (developers side). Is my understanding corretct?
In that case, even If we commit `Cargo.lock`, we'll have to test our crate with latest dependencies by running `cargo update & cargo build` after all.  I'm not exactly sure if how much useful `Cargo.lock` for library is other than making the detection of regression easier...

**Reply by rusty-snake** [reply]
Yes, debugging (git bisect, ...) and testing of dependencies are the main points here. The latest dependencies CI problem is not relevant if you update them with dependabot. We do not use much bisect (for dependency issues) and as long as the MSRV job does not require action in that direction, I don't see much reason to commit it.

**Reply by ManaSugi** [reply]
Okay, thanks. I don't have strong opinion about this, but I assume there is no any disadvantage we commit `Cargo.lock`. 
PRs will be sometimes created to update `Cargo.lock` by dependabot, but we can also configure to make the PRs merged automatically when our CI passes, so there is no additional workload on our side. Besides, we can solve dependencies issues earlier with isolating the issue between our own codes and external crates.
We have encountered the ugly issue about musl-libc recently (the issue seems to be solved silently now). I think deterministic build would help us do debug such a issue. WDYH?

**Comment by rusty-snake** [comment]
Update: The general recommendation is now to commit Cargo.lock.

I guess we should go that way.

**Comment by ManaSugi** [comment]
we might want to push `Cargo.lock`
